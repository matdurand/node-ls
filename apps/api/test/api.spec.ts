/**
 * @group e2e/api
 */
import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { LsModule } from '../src/ls.module';

describe('LsController (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [LsModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  it('/folder-content (GET)', () => {
    return request(app.getHttpServer())
      .get('/folder-content?path=testfolder')
      .expect(200)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });
});
