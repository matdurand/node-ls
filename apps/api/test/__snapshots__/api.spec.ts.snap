// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LsController (e2e) /folder-content (GET) 1`] = `
Object {
  "filesCount": 3,
  "items": Array [
    Object {
      "lastModified": "2020-12-22T19:53:21.834Z",
      "name": "folder01",
      "size": 0,
      "type": "folder",
    },
    Object {
      "lastModified": "2020-12-22T19:53:21.834Z",
      "name": "folder1",
      "size": 0,
      "type": "folder",
    },
    Object {
      "lastModified": "2020-12-22T19:53:21.834Z",
      "name": "aa.txt",
      "size": 2,
      "type": "file",
    },
    Object {
      "lastModified": "2020-12-22T19:53:21.834Z",
      "name": "b.txt",
      "size": 18,
      "type": "file",
    },
    Object {
      "lastModified": "2020-12-22T19:53:21.830Z",
      "name": "a.txt",
      "size": 929,
      "type": "file",
    },
  ],
  "subfoldersCount": 2,
  "totalSize": 949,
}
`;
